### PR TITLE
Fix error in mixed precision training of `TFCvtModel`

### PR DESCRIPTION
**Pull request by gcuder**
# What does this PR do?

This PR fixes the issue that the `TFCvtModel` cannot be trained with `keras.fit` using `mixed-precision`. 
The issue was in this [line](https://github.com/huggingface/transformers/blob/c4bf6f38bda1de3798095515875a119298bf0611/src/transformers/models/cvt/modeling_tf_cvt.py#L96) when a random tensor is initialized without specifying the correct `dtype`.  

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@amyeroberts

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for adding and including a test ❤️ 

Let's get a review from @Rocketknight1 before merging to make sure it's in line with TF design

File: tests/models/cvt/test_modeling_tf_cvt.py
Comment (Rocketknight1) [inline tests/models/cvt/test_modeling_tf_cvt.py:192]:
```suggestion
        tf.keras.mixed_precision.set_global_policy(policy)
        super().test_keras_fit()
        tf.keras.mixed_precision.set_global_policy("float32")
```

**Review by Rocketknight1 — APPROVED** [review]
This looks good to me, and I think using `self.compute_dtype` is fine as long as tests pass. My one concern is that using `set_global_policy` is a framework-level thing that might (maybe?) affect other tests - just to be safe, let's set it back afterwards.

**Pull request merged**
